src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/AuditServiceTest.kt

```diff
@@ -0,0 +1,198 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.service
+
+import com.nhaarman.mockitokotlin2.argThat
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import org.junit.jupiter.api.Test
+import org.mockito.ArgumentMatcher
+import org.springframework.context.annotation.Bean
+import org.springframework.test.util.ReflectionTestUtils
+import uk.gov.justice.digital.hmpps.pecs.jpc.auditing.AuditEvent
+import uk.gov.justice.digital.hmpps.pecs.jpc.auditing.AuditEventRepository
+import uk.gov.justice.digital.hmpps.pecs.jpc.auditing.AuditEventType
+import uk.gov.justice.digital.hmpps.pecs.jpc.config.TimeSource
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.LocationType
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.Money
+import java.time.LocalDateTime
+
+private class AuditEventMatcher(var auditEvent: AuditEvent) : ArgumentMatcher<AuditEvent> {
+  override fun matches(otherAuditEvent: AuditEvent): Boolean {
+    return auditEvent.username == otherAuditEvent.username &&
+      auditEvent.details == otherAuditEvent.details &&
+      auditEvent.eventType == otherAuditEvent.eventType
+  }
+}
+
+internal class AuditServiceTest {
+  private val auditEventRepository: AuditEventRepository = mock()
+
+  @Bean
+  private fun timeSource() = TimeSource { LocalDateTime.now() }
+
+  @Test
+  internal fun `create log in audit event`() {
+    val service = AuditService(auditEventRepository)
+    ReflectionTestUtils.setField(service, "timeSource", timeSource())
```

Comment:
We don't need to this `ReflectionTestUtils.....`.  Probably just east to talk about it.